server/services/lists/caCaseloadService.ts

```diff
@@ -521,4 +425,30 @@ export default class CaCaseloadService {
       return view === 'prison' ? crd1 - crd2 : crd2 - crd1
     })
   }
+
+  splitCasesByComDetails(cases: CaCase[]) {
+    const groupedCases = cases.reduce((acc, c) => {
+      const groups = acc
+      if (!groups.staffCode) {
```

Comment:
**suggestion:** rather than doing this on each iteration would just initialise your `acc`?

`{ staffCode = [],  staffUsername = [] ...etc.. } as GroupedByCom`

also should those be pluralized?

Reply:
Goddamn it I looked for ages to see if I could set a default/initial value for `acc` and found nothing. I was surprised that I seemingly couldn't, but turns out that's because I can 😅 

To be honest I'm thinking that I don't like the names anyway, as it implies that the array is a list of `staffCode`s, `staffUsername`s etc, whereas it's actually cases _with_ a staff code or _with_ a staff username.
I might just rename them while I'm changing it anyway